src/domain/FineractClient/FineractClient.ts

```diff
@@ -25,275 +25,232 @@
+                'FIN',
+            );
+        } else if (!getAccountRes.data.status.active) {
+            throw new FineractAccountNotActiveError(
+                `Fineract Account not active ${JSON.stringify(getAccountRes)}`,
+                'FIN',
+            );
+        }
+        const currency = getAccountRes.data.currency.code;
+        const getClientRes = await this.getClient(getAccountRes.data.clientId);
+        if (getClientRes.statusCode != 200) {
+            throw new FineractGetClientWithIdError(
+                `Failed to get client with Id ${getAccountRes.data.clientId}`,
+                'FIN',
+            );
+        }
+        return {
+            accountId: returnedEntity.entityId,
+            data: getClientRes.data,
+            status: getClientRes.statusCode,
+            currency: currency,
+        };
     }
 
-    private async searchAccount(accountNo: string): Promise<THttpResponse<TFineractSearchResponse> | undefined>{
+    private async searchAccount(accountNo: string): Promise<THttpResponse<TFineractSearchResponse>> {
         const url = `${this.fineractConfig.FINERACT_BASE_URL}/${ROUTES.search}?query=${accountNo}&resource=savingsaccount`;
         this.logger.info(`Request to fineract ${url}`);
-        const res = await this.httpClient.send<TFineractSearchResponse>(
-            url,
-            {
-                method: "GET",
-                headers: {
-                    "fineract-platform-tenantId": this.fineractConfig.FINERACT_TENTANT_ID,
-                    "Authorization": this.getAuthHeader()
-                }
-            }
-        );
+        const res = await this.httpClient.send<TFineractSearchResponse>({
+            url: url,
+            method: 'GET',
+            headers: {
+                'fineract-platform-tenantId': this.fineractConfig.FINERACT_TENANT_ID,
+                Authorization: this.getAuthHeader(),
+            },
+        });
         return res;
     }
 
-    private async getSavingsAccount(accountId: number): Promise<THttpResponse<TFineractGetAccountResponse> | undefined>{
+    async getSavingsAccount(accountId: number): Promise<THttpResponse<TFineractGetAccountResponse>> {
         const url = `${this.fineractConfig.FINERACT_BASE_URL}/${ROUTES.savingsAccount}/${accountId}`;
-        this.logger.info(`Request to fineact ${url}`);
-        const res = await this.httpClient.send<TFineractGetAccountResponse >(
-            url,
-            {
-                method: "GET",
-                headers: {
-                    "fineract-platform-tenantId": this.fineractConfig.FINERACT_TENTANT_ID,
-                    "Authorization": this.getAuthHeader()
-                }
-            }
-        );
+        this.logger.info(`Request to fin    eact ${url}`);
+        const res = await this.httpClient.send<TFineractGetAccountResponse>({
+            url: url,
+            method: 'GET',
+            headers: {
```

Comment:
this headers is copied/pasted (DRY).  Can't we put this logic in one place and  reuse it everywhere?